### ISSUE TITLE
Forge fails to fork against Anvil instances that load state

### ISSUE THREAD

**Issue by emo-eth** [foundry-rs/foundry#7284]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [ ] Foundryup

### What command(s) is the bug in?

forge test

### Operating System

macOS (Apple Silicon)

### Describe the bug

When anvil is loaded with a `--load-state` parameter, Forge fails with the error:

```
2024-02-29T21:50:20.568272Z ERROR foundry_evm_core::fork::init: It looks like you're trying to fork from an older block with a non-archive node which is not supported. Please try to change your RPC url to an archive node if the issue persists.
```
when the anvil RPC is specified as a fork url.

**Comment by ArshanKhanifar** [comment]
I'm having the same issue.

**Comment by chrismin** [comment]
I'm having the same issue as well.

**Comment by qt-ritual** [comment]
me three! 
![Uploading Screenshot 2024-06-28 at 7.18.15 PM.png…]()

**Comment by qt-ritual** [comment]
me three!
<img width="545" alt="Screenshot 2024-06-28 at 7 18 15 PM" src="https://github.com/foundry-rs/foundry/assets/167802755/40b6f031-8c67-4cd7-b96c-92cc5083548e">

**Comment by mattsse** [comment]
this error is logged if the server responds with "missing trie node" error which means it does not have the requested state to fork from.
please try another endpoint

**Comment by emo-eth** [comment]
@zerosnacks seems to have been resolved by #7502
